src/09_output_housing_value.R

```diff
@@ -0,0 +1,461 @@
+  #       ) <
+  #         1e-6,
+  #       0,
+  #       1
+  #     ),
+  #     na_gross_improvement_value = if_else(
+  #       is.na(GEN_GROSS_IMPROVEMENT_VALUE),
+  #       1,
+  #       0
+  #     ),
+  #     na_net_improvement_value = if_else(
+  #       is.na(GEN_NET_IMPROVEMENT_VALUE),
+  #       1,
+  #       0
+  #     ),
+  #     diff_land_value = if_else(
+  #       abs(
+  #         GEN_GROSS_LAND_VALUE - GEN_NET_LAND_VALUE
+  #       ) <
+  #         1e-6,
+  #       0,
+  #       1
+  #     ),
+  #     na_gross_land_value = if_else(
+  #       is.na(GEN_GROSS_LAND_VALUE),
+  #       1,
+  #       0
+  #     ),
+  #     na_net_land_value = if_else(
+  #       is.na(GEN_NET_LAND_VALUE),
+  #       1,
+  #       0
+  #     )
+  #   ) %>%
+  #   summarise(
+  #     across(
+  #       .cols = starts_with("na") | starts_with("diff"),
+  #       .fns = ~ mean(.x, na.rm = T)
+  #     )
+  #   ) %>%
+  #   # filter(!diff_improvement_value | !diff_land_value) %>%
+  #   collect()
+
+  # na_gross_improvement_value	na_net_improvement_value	na_gross_land_value	na_net_land_value	diff_improvement_value	diff_land_value
+  # 0	0.073751	0	0.073751	0.124272	0.075824
+
+  # bca_properties %>%
+  #   filter(is.na(GEN_NET_IMPROVEMENT_VALUE)) %>%
+  #   glimpse()
+
+  bca_properties <- bca_properties %>%
+    mutate(
+      GEN_GROSS_IMPROVEMENT_VALUE = as.numeric(GEN_GROSS_IMPROVEMENT_VALUE),
+      GEN_GROSS_LAND_VALUE = as.numeric(GEN_GROSS_LAND_VALUE)
+    ) %>%
+    dplyr::filter(
+      GEN_PROPERTY_SUBCLASS_DESC %in%
+        c('Residential Single Family', 'Residential Strata'),
+      GEN_GROSS_IMPROVEMENT_VALUE > 50000,
+      GEN_GROSS_LAND_VALUE > 50000
+    ) %>%
+    mutate(
+      YR = year_name, # Use the year from our loop
+      TOTAL_VALUE = GEN_GROSS_IMPROVEMENT_VALUE + GEN_GROSS_LAND_VALUE
+    )
+
+  # Join the tables to create folio data for this year
+  folio_data <- bca_properties %>%
+    left_join(bca_addrs, by = "FOLIO_ID") %>%
+    left_join(gcs_data, by = c("PC" = "POSTALCODE"))
+
+  # Step 3: Calculate average property statistics by DA and jurisdiction
+  ave_p <- folio_data %>%
+    group_by(YR, DA) %>%
+    summarize(
+      N_PROPERTY = n_distinct(FOLIO_ID),
+      AVE_IMPROVEMENT_VALUE = mean(GEN_GROSS_IMPROVEMENT_VALUE, na.rm = TRUE),
+      AVE_LAND_VALUE = mean(GEN_GROSS_LAND_VALUE, na.rm = TRUE),
+      AVE_TOTAL_VALUE = mean(TOTAL_VALUE, na.rm = TRUE),
+      MIN_TOTAL_VALUE = min(TOTAL_VALUE, na.rm = TRUE),
+      MAX_TOTAL_VALUE = max(TOTAL_VALUE, na.rm = TRUE),
+      .groups = "drop"
+    )
+
+  ave_p <- ave_p %>%
+    collect()
+
+  # ave_p %>% count(DA) %>% filter(n > 1)
+  # ave_p %>%
+  #   filter(DA == "010120219")
+  # 1063 DAs in 2023 has at least two JURISDICTIONs. It causes another problem.
+
+  # Step 4: Calculate median property values: median function is not available in dbplyr with group by and summarise, so we need to use mutate and distinct
+  med_p <- folio_data %>%
```

Comment:
I think this can all be done in one step now? It makes the code unnecessarily complex calculating the median separately IMO

Reply:
never mind, dbplyr is silly as discussed